source/s3_meta_request.c

```diff
@@ -1385,14 +1398,16 @@ static void s_s3_meta_request_stream_complete(struct aws_http_stream *stream, in
     if (meta_request->checksum_config.validate_response_checksum) {
         s_get_response_part_finish_checksum_helper(connection, error_code);
     }
-    if (error_code != AWS_ERROR_S3_CANCELED && error_code != AWS_ERROR_S3_PAUSED) {
-        /* BEGIN CRITICAL SECTION */
+    /* BEGIN CRITICAL SECTION */
+    {
         aws_s3_meta_request_lock_synced_data(meta_request);
-        AWS_ASSERT(request->synced_data.http_stream != NULL);
-        aws_linked_list_remove(&request->ongoing_http_requests_list_node);
+        if (request->synced_data.ongoing_http_stream) {
+            /* The request has been cancelled, and the node has been removed. */
```

Comment:
this comment is confusing. the request is complete but it wasn't necessarily canceled.
and we're about to remove the node, but the comment make it sound like it was already removed

delete the comment entirely, the code makes enough sense now, or replace with something like
```suggestion
            /* Remove from list of ongoing requests */
```

Reply:
yeah, updated the code, but I forgot to update the comment around it ...